src/relations/postgresql_provider.py

```diff
@@ -119,7 +119,8 @@ def _on_relation_broken(self, event: RelationBrokenEvent) -> None:
         """Remove the user created for this relation."""
         # Check for some conditions before trying to access the PostgreSQL instance.
         if (
-            "cluster_initialised" not in self.charm._peers.data[self.charm.app]
+            not self.charm._peers
```

Comment:
When units departed, this check fails due to `_peers` being None, leaving the unit stuck in removing in Juju 3.